lib/Models/CartoMapCatalogItem.ts

```diff
@@ -10,33 +11,39 @@ import UrlTemplateImageryProvider from "terriajs-cesium/Source/Scene/UrlTemplate
 import Resource from "terriajs-cesium/Source/Core/Resource";
 import Rectangle from "terriajs-cesium/Source/Core/Rectangle";
 import isDefined from "../Core/isDefined";
-import CommonStrata from "./CommonStrata";
-
-export default class CartoMapCatalogItem
-  extends UrlMixin(CatalogMemberMixin(CreateModel(CartoMapCatalogItemTraits)))
-  implements Mappable {
-  static readonly type = "carto";
-
-  get type() {
-    return CartoMapCatalogItem.type;
+import LoadableStratum from "./LoadableStratum";
+import StratumOrder from "./StratumOrder";
+import { BaseModel } from "./Model";
+
+export class CartoLoadableStratum extends LoadableStratum(
+  CartoMapCatalogItemTraits
+) {
+  static stratumName = "cartoLoadable";
+
+  constructor(
+    readonly catalogItem: CartoMapCatalogItem,
+    readonly tileUrl: string,
+    readonly tileSubdomains: string[]
+  ) {
+    super();
   }
 
-  get isMappable() {
-    return true;
+  duplicateLoadableStratum(newModel: BaseModel): this {
+    return new CartoLoadableStratum(
+      newModel as CartoMapCatalogItem,
+      this.tileUrl,
+      this.tileSubdomains
+    ) as this;
   }
 
-  loadMapItems(): Promise<void> {
-    if (isDefined(this.tileUrl)) {
-      return Promise.resolve();
-    }
-
+  static load(catalogItem: CartoMapCatalogItem): Promise<CartoLoadableStratum> {
     let queryParameters: { auth_token?: string };
     queryParameters = {};
-    if (this.auth_token) {
-      queryParameters.auth_token = this.auth_token;
+    if (catalogItem.auth_token) {
+      queryParameters.auth_token = catalogItem.auth_token;
     }
 
-    if (this.url === undefined) {
+    if (catalogItem.url === undefined) {
       throw new TerriaError({
```

Comment:
This throw should be promise rejection (`Promise.reject(error)` instead of `throw errror`)